rosie/src/main/java/com/karumi/rosie/domain/usecase/UseCaseHandler.java

```diff
@@ -29,6 +33,10 @@
   private final TaskScheduler taskScheduler;
   private final ErrorHandler errorHandler;
 
+  // This list is used to retain error adapters in a non-weak reference associated with this use
+  // case handler
+  private final List<OnErrorCallback> errorCallbacks = new ArrayList<>();
```

Comment:
I'm not sure if in that case there may be a problem, UseCaseHandler in RosiePresenter is not a weak reference, but I think all dependencies depend on the activity.
Maybe, we can do how in RosiePresenter the globalOnErrorCallbacks list is cleared in destroy method. In destroy method we can call to UseCaseHandler to clear the new list, but if we do it, we should check the null pointer in the WeakReference in RosieUseCase.notifyError()